app/src/main/java/com/nervousfish/nervousfish/activities/MainActivity.java

```diff
@@ -207,6 +225,49 @@ private void fillDatabaseWithDemoData() throws IOException {
     }
 
     /**
+     * Called when a new contact is received
+     *
+     * @param event Contains additional data about the event
+     */
+    @Subscribe(threadMode = ThreadMode.MAIN)
+    public void onNewDataReceivedEvent(final NewDataReceivedEvent event) {
+        if (event.getClazz().equals(Contact.class)) {
+            final Contact contact = (Contact) event.getData();
+            try {
+                LOGGER.info("Checking if the contact exists...");
+                if (checkExists(contact)) {
+                    LOGGER.warn("Contact already existed...");
+                } else {
+                    LOGGER.info("Adding contact to database...");
+                    this.serviceLocator.getDatabase().addContact(contact);
+                    this.contacts = this.serviceLocator.getDatabase().getAllContacts();
+                    sortOnName();
```

Comment:
Well, ehm, no... I didn't test it so I'm not sure if it works. I did test the Bluetooth communication and that seems to work quite well. I propose that we merge this PR asap so that we can finish on time the rhythm pairing. When it's merged we can let somebody fix this issue (we cannot do it earlier than the next meeting at Wednesday).